lib/perl/Genome/Config.pm

```diff
@@ -0,0 +1,158 @@
+package Genome::Config;
+
+use strict;
+use warnings;
+
+use File::Find::Rule qw();
+use Genome::Carp qw(croakf);
+use Genome::ConfigSpec qw();
+use Path::Class qw();
+use YAML::Syck qw();
+
+=item get()
+
+C<get($key)> retrieves the configuration value specified by the input key.  It
+does a hierarchival search for the value,
+
+- environment variable
+- user config file
+- snapshot config file
+- global config files
+
+User's config files are discovered in C<XGENOME_CONFIG_HOME> which defaults to
+C<$HOME/.config>.  Global config files are discovered in C<XGENOME_CONFIG_DIRS>
+which defaults to C</etc> and can be a colon-delimited list of directories like
+C<PATH>.
+
+Values are specified in C<genome/config.yaml> in one of the source directories
+or via the corresponding environment variable.
+
+Configuration values are "registered" by creating a spec file, see
+L<Config::Spec>.  Spec files can go in either the snapshot config directory or
+the global config directories in the C<genome> subdirectory, e.g.
+C</etc/genome/foo.yaml>.
+
+If you want the configuration value to be optional you must specify a
+C<default_value> even if it is zero or an empty string but it must be defined.
```

Comment:
This is a run-on sentence.  One possibility is to split it in two: `...specify a C<default_value>.  The default can be zero or an empty string, but it must be defined.`